src/core/evaluator.js

```diff
@@ -4837,8 +4845,7 @@ class EvaluatorPreprocessor {
             // used to error, rather than just warn, once a number of invalid
             // path operators have been encountered (fixes bug1443140.pdf).
             if (
-              fn >= OPS.moveTo &&
-              fn <= OPS.endPath && // Path operator
+              this._isPathOp &&
               ++this._numInvalidPathOPS >
                 EvaluatorPreprocessor.MAX_INVALID_PATH_OPS
             ) {
```

Comment:
Maybe we could remove this `throw` and just warn that the page rendering could be wrong because of incomplete operators.
Do you know if Acrobat (or pdfium-based viewers) has such a threshold (i.e. `MAX_INVALID_PATH_OPS`) to decide to "invalidate" a pdf ?

Reply:
> Maybe we could remove this `throw` and just warn that the page rendering could be wrong because of incomplete operators.

That's what we used to do, but if you look at e.g. `bug1443140.pdf` *without* this heuristic you'll see why it's done :-)

> Do you know if Acrobat (or pdfium-based viewers) has such a threshold (i.e. `MAX_INVALID_PATH_OPS`) to decide to "invalidate" a pdf ?

No idea unfortunately; and how different PDF viewer deal with corrupt data seem (as far as I can tell) to vary a fair bit...

Reply:
Ok, it makes sense.
Just for information, it seems that pdfium (and the same in Preview on mac) doesn't stop to parse: in Chrome, on page 4 of `bug1443140.pdf` in the bottom left there are some garbage text which is at the end of the content stream.
Anyway as said it just for information and I don't think it's worth spending so much time to fix that stuff so r=me.